utils/utils.go

```diff
@@ -143,3 +144,15 @@ func ExtractAndMaskIP(ipStr string) (maskedIP string, ok bool) {
 		return ApplyIPMask(ipStr)
 	}
 }
+
+func ExtractVersionAndServiceFromUserAgent(userAgent string) (reqVer, service string) {
```

Comment:
Can you add a top-level comment to the function that describes its intended use along with the contents of the outputs (i.e. what reqVer and service look like when they're returned)? Also, can you add a unit test with any edge cases you can think of?